Libraries/LibC/pwd.cpp

```diff
@@ -125,4 +125,43 @@ struct passwd* getpwent()
     strncpy(__pwdb_entry->shell_buffer, e_shell.characters(), PWDB_STR_MAX_LEN);
     return __pwdb_entry;
 }
+
+int putpwent(const struct passwd* p, FILE* stream)
+{
+    // glibc does some more advanced checks on p's members. We should too eventually.
+    // For now, checking for NULL and setting EINVAL should be good enough :)
+    if (!p || !stream || !p->pw_name || !p->pw_dir || !p->pw_gecos || !p->pw_shell) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (strchr(p->pw_name, ':') || strchr(p->pw_name, '\n')) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (strchr(p->pw_dir, ':') || strchr(p->pw_dir, '\n')) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (strchr(p->pw_gecos, ':') || strchr(p->pw_gecos, '\n')) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (strchr(p->pw_shell, ':') || strchr(p->pw_shell, '\n')) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    // Looks like we didn't write any characters at all!
```

Comment:
This comment should be inside the error handling block, instead of before the error has occurred.